pages/connect.tsx

```diff
@@ -0,0 +1,135 @@
+import { PermissionType } from 'arconnect';
+import React, { useState, useEffect } from 'react';
+import Router from 'next/router';
+import useGetUser from '../hooks/useGetUser';
+
+const Connect = () => {
+  const [userName, setUserName] = useState(''),
+    [allUserNames, setAllUserNames] = useState<string[]>([]),
+    { currentUser } = useGetUser();
+
+  useEffect(() => {
+    const getAllUsers = async () => {
+      try {
+        const response = await fetch('/api/getAllPosts');
+        const getPosts = await response.json();
+        const { users } = getPosts.data;
+
+        users.map((user: { walletAddress: string; userName: string }) => {
+          setAllUserNames(() => [...allUserNames, user.userName]);
```

Comment:
Waaaayyyy too many calls and array constructions. Bad bad
Construct an array 
```
const users = users.map((user) => user.userName);
```
then call the hook
```
setAllUserNames(users)
```